test/runtests.jl

```diff
@@ -458,83 +458,6 @@ end
     @test _x509_certificate.issuer_name == x509_certificate.issuer_name
 end
 
-# https://www.openssl.org/docs/man3.0/man7/OSSL_PROVIDER-legacy.html
-@testset "Encrypt" begin
-    if OpenSSL.version_number() ≥ v"3"
-        OpenSSL.load_legacy_provider()
```

Comment:
The `load_legacy_provider` function and associated code would go away, I suppose.